server/src/main/java/com/tencent/rss/server/ShuffleServerGrpcService.java

```diff
@@ -426,21 +437,63 @@ public void getShuffleData(GetShuffleDataRequest request,
     responseObserver.onCompleted();
   }
 
-  private List<ShuffleDataBlockSegment> toBlockSegments(List<BufferSegment> bufferSegments) {
-    List<ShuffleDataBlockSegment> ret = Lists.newArrayList();
-
-    for (BufferSegment segment : bufferSegments) {
-      ret.add(ShuffleDataBlockSegment.newBuilder()
-          .setBlockId(segment.getBlockId())
-          .setOffset(segment.getOffset())
-          .setLength(segment.getLength())
-          .setUncompressLength(segment.getUncompressLength())
-          .setCrc(segment.getCrc())
-          .setTaskAttemptId(segment.getTaskAttemptId())
-          .build());
+  @Override
+  public void getShuffleIndex(GetShuffleIndexRequest request,
+      StreamObserver<GetShuffleIndexResponse> responseObserver) {
+    ShuffleServerMetrics.counterTotalRequest.inc();
+
+    String appId = request.getAppId();
+    int shuffleId = request.getShuffleId();
+    int partitionId = request.getPartitionId();
+    int partitionNumPerRange = request.getPartitionNumPerRange();
+    int partitionNum = request.getPartitionNum();
+    StatusCode status = StatusCode.SUCCESS;
+    String msg = "SUCCESS";
+    GetShuffleIndexResponse reply;
+    String requestInfo = "appId[" + appId + "], shuffleId[" + shuffleId + "], partitionId["
+        + partitionId + "]";
+
+    if (shuffleServer.isMultiStorageEnabled()) {
+      shuffleServer.getMultiStorageManager().updateLastReadTs(appId, shuffleId, partitionId);
     }
 
-    return ret;
+    long indexFileSize = shuffleServer.getShuffleServerConf().getLong(ShuffleServerConf.SERVER_SHUFFLE_INDEX_SIZE_HINT);
```

Comment:
indexFileSize -> assumedFileSize
there should be comment for why this is used and how many blockIds can be supported with default value

Reply:
OK